File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int n  = nums.size();
        vector<int> a(n,1),b(n,1);
        for(int i =1;i<nums.size();i++){
            // a.push_back(1)
            a[i] = nums[i-1]*a[i-1];
        }
        for(int i = n-2;i>=0;i--){
            // a.push_back(1)
            b[i] = nums[i+1]*b[i+1];
        }
        vector<int> res(n,1);
        for(int i=0;i<n;i++){
            res[i] = a[i]*b[i];
        }
        return res;
    }
};

File: Data Structures & Algorithms/three-integer-sum/submission-0.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        i, j = 0,0
        out = []
        for i in range(len(nums)):
            target = 0 - nums[i]
            temp = {}
            res = []
            for j in range(i+1, len(nums)):
                if target - nums[j] not in temp:
                    temp[nums[j]] = j
                else:
                    res.append(nums[i])
                    res.append(nums[j])
                    res.append(target-nums[j])
                    res.sort()
                    if res not in out:
                        out.append(res)
                    res = []
                    # break
                j+=1
            i+=1
            # out.append(res)
        
        return out

        